src/flowportApi/request.ts

```diff
@@ -0,0 +1,94 @@
+export class HttpRequestError extends Error {
+  readonly httpStatus: number
+  readonly headers: Record<string, string>
+  readonly responseText: string
+  readonly url: string
+  readonly method: string
+
+  constructor(
+    msg: string,
+    httpStatus: number,
+    headers: Record<string, string>,
+    responseText: string,
+    url: string,
+    method: string,
+  ) {
+    super(msg)
+    this.httpStatus = httpStatus
+    this.headers = headers
+    this.responseText = responseText
+    this.url = url
+    this.method = method
+  }
+}
+
+type RequestParams = {
+  url: string
+  method?: string
+  body?: string | null
+  headers?: Record<string, string>
+  parseResponse?: boolean
+  timeout?: number
+}
+
+export async function request<T = unknown>({
```

Comment:
I'd consider using something more standardized for making fetch requests, like `axios`

Reply:
It is in a separate file, but its in the flowport folder, as I considered it an util of the flowport api